test/jdk/java/foreign/LibraryLookupTest.java

```diff
@@ -91,6 +97,37 @@ void testLoadLibraryBadLookupName() {
         }
     }
 
+    @Test
+    void testLoadLibraryNonDefaultFileSystem() throws URISyntaxException, IOException {
+        try (FileSystem customFs = fsFromJarOfClass(org.testng.annotations.Test.class)) {
+            try (Arena arena = Arena.ofConfined()) {
+                Path p = customFs.getPath(".");
+                try {
+                    SymbolLookup.libraryLookup(p, arena);
+                    fail("Did not throw IAE");
+                } catch (IllegalArgumentException iae) {
+                    assertTrue(iae.getMessage().contains("not in default file system"));
+                }
+            }
+        }
+    }
+
+    private static FileSystem fsFromJarOfClass(Class<?> clazz) throws URISyntaxException, IOException {
+        String name = clazz.getName();
+        final int lastDot = name.lastIndexOf('.');
+        if (lastDot != -1) {
+            name = name.substring(lastDot + 1);
+        }
+        URI uri = clazz.getResource(name + ".class").toURI();
+        if (uri.getScheme().equals("jar")) {
+            final String[] parts = uri.toString().split("!");
+            if (parts.length == 2) {
+                return FileSystems.newFileSystem(URI.create(parts[0]), new HashMap<>());
+            }
+        }
+        throw new AssertionError("Unable to create file system from " + clazz);
```

Comment:
Using a class from testng.jar is a good idea. An alternative to decoding the composite URL is to just use the file path to the JAR file, e.g.
```
URI uri = clazz.getProtectionDomain().getCodeSource().getLocation().toURI();
return FileSystems.newFileSystem(Path.of(uri));
```